.github/actions/set-container-repo-and-determine-image-tag/action.yml

```diff
@@ -58,3 +64,30 @@ runs:
         fi
 
         echo "image-tag=${image_tag}" >> $GITHUB_OUTPUT
+    - name: Output gadgets repository
+      id: gadgets-repo
+      shell: bash
+      run: |
+        if [ -z '${{ inputs.registry }}' ]; then
+          echo 'inputs.registry should not be empty! Set it before calling this action' 1>&2
+          exit 1
+        fi
+
+        gadgets_repo=${{ inputs.registry }}/${{ github.repository_owner }}
+        if [ ${{ github.event_name }} == 'pull_request' ]; then
+          gadgets_repo=ttl.sh/${{ github.sha }}
+        fi
+
+        echo "gadgets-repo=${gadgets_repo}" >> $GITHUB_OUTPUT
+    - name: Output gadgets image tag
+      id: gadgets-tag
+      shell: bash
+      run: |
+        # If GITHUB_REF_NAME is 'foo/bar', image_tag will be 'foo-bar', we need
+        # this because it is not possible to have slash in image tag.
+        gadgets_tag=${GITHUB_REF_NAME//\//-}
+        if [ "$gadgets_tag" = "main" ]; then
+            gadgets_tag="latest"
```

Comment:
I don't know the convention about the `latest` tag.
As an end user I always think its the latest release version. Maybe something like `head` or `dev` is better. On release we tag it with `latest`.

Another (maybe) issue:
What if I call my branch `latest` or some version number `v0.20.0`

Reply:
> As an end user I always think its the latest release version. Maybe something like head or dev is better. 

AFAUI `latest` is used when no tag is specified, I'm not sure there is a standard convention to indicate it's the latest tagged release. 

>On release we tag it with latest.


Using `latest` isn't recommended , users should stick to a fixed version, so I think we can continue using latest for development. 

> What if I call my branch latest or some version number v0.20.0

I think there are other cases where this can get broken, but IMO they're not so relevant. We're following the same approach for other Inspektor Gadget images.